Orleans.Providers.MongoDB/StorageProviders/Serializers/Configuration/IStateProviderSerializerOptions.cs

```diff
@@ -0,0 +1,10 @@
+﻿namespace Orleans.Providers.MongoDB.StorageProviders.Serializers.Configuration
+{
+    public interface IStateProviderSerializerOptions
```

Comment:
What is this interface needed for?

Reply:
It's supposed to serve as a common interface for all grain storage providers that need GrainStateSerializer, but at a second look, we only have one storage provider, so I was able to remove it.
I've replaced it with `MongoDBGrainStorageConfigurator` which sets `GrainStateSerializer` value to the one registered in the DI under the same name as the provider.